src/CachePlugin.php

```diff
@@ -69,8 +70,12 @@ final class CachePlugin implements Plugin
      *              Defaults to an empty array
      * }
      */
-    public function __construct(CacheItemPoolInterface $pool, StreamFactory $streamFactory, array $config = [])
+    public function __construct(CacheItemPoolInterface $pool, $streamFactory, array $config = [])
     {
+        if (!($streamFactory instanceof StreamFactory) && !($streamFactory instanceof StreamFactoryInterface)) {
+            throw new \TypeError(\sprintf('StreamFactory must be an instance of %s or %s.', StreamFactory::class, StreamFactoryInterface::class));
```

Comment:
Should be:

```
throw new \TypeError(\sprintf('Argument 2 passed to %s::__construct() must be of type %s|%s, %s given.', self::class, StreamFactory::class, StreamFactoryInterface::class, gettype($streamFactory)));
```

To match the message given by PHP 8 for type errors with union types.

Reply:
Even better: `\is_object($streamFactory) ? \get_class($streamFactory) : \gettype($streamFactory)` for the given type ;)

Reply:
Ah, yes. 👍